TombEngine/Objects/TR5/Entity/tr5_lion.cpp

```diff
@@ -118,94 +142,98 @@ namespace TEN::Entities::Creatures::TR5
 						break;
 					}
 
-					if (AI.ahead)
+					if (targetAhead && AI.bite)
 					{
-						if (item->TouchBits.Test(LionAttackJoints))
-						{
-							item->Animation.TargetState = LION_STATE_BITE_ATTACK;
-							break;
-						}
-
-						if (AI.distance < LION_POUNCE_ATTACK_RANGE)
+						if (AI.distance >= LION_ATTACK_MIN_RANGE)
 						{
-							item->Animation.TargetState = LION_STATE_POUNCE_ATTACK;
-							break;
+							if (AI.distance <= LION_BITE_ATTACK_RANGE)
+							{
+								item->Animation.TargetState = LION_STATE_BITE_ATTACK;
+								break;
+							}
+							if (AI.distance < LION_POUNCE_ATTACK_RANGE)
+							{
+								item->Animation.TargetState = LION_STATE_POUNCE_ATTACK;
+								break;
+							}
 						}
 					}
 
 					item->Animation.TargetState = LION_STATE_RUN_FORWARD;
-					break;
+				break;
 
 				case LION_STATE_WALK_FORWARD:
-					creature->MaxTurn = ANGLE(2.0f);
+					creature->MaxTurn = LION_WALK_TURN_RATE_MAX;
 
 					if (creature->Mood == MoodType::Bored)
 					{
-						if (TestProbability(1.0f / 256))
-						{
-							item->Animation.TargetState = LION_STATE_IDLE;
-							item->Animation.RequiredState = LION_STATE_ROAR;
-						}
+						if (TestProbability(LION_ROAR_CHANCE))
```

Comment:
Move this to lign 168 `
if (creature->Mood == MoodType::Bored && TestProbability(LION_ROAR_CHANCE)) {
item->Animation.TargetState = LION_STATE_ROAR;
}
`

Reply:
Ohm true, I didn't notice that.

Reply:
done